### PR TITLE
[icu] Fix configure failure due to not finding python

### PR DESCRIPTION
**Pull request by LilyWangL**
**Describe the pull request**
Fix configure failure due to not finding python.
- What does your PR fix? Fixes issue #10642

- No features need to test.

### CODE REVIEW

File: ports/icu/portfile.cmake [hunk 1]
```diff
@@ -24,6 +22,11 @@ vcpkg_extract_source_archive_ex(
         ${CMAKE_CURRENT_LIST_DIR}/fix-extra.patch
 )
 
+vcpkg_find_acquire_program(PYTHON3)
+get_filename_component(PYTHON3_DIR "${PYTHON3}" DIRECTORY)
+vcpkg_add_to_path("${PYTHON3_DIR}")
```
Comment (Neumann-A) [inline ports/icu/portfile.cmake:27]: Could be that those two lines are unnecessary. I just added them in as the first try and it did not work. I then  added `set(ENV{PYTHON} "${PYTHON3}")` which really solved the problem. 
The failure is very likely related to the fact that Win10 tends to redirect the python call to the Windows AppStore if python is not installed.

Comment (Neumann-A) [reply ports/icu/portfile.cmake:27]: Just test it:
```
vcpkg_find_acquire_program(PYTHON3)
set(ENV{PYTHON} "${PYTHON3}")
```
is enough to make it work

Comment (LilyWangL) [reply ports/icu/portfile.cmake:27]: Thanks for your help~

**Comment by Voskrese** [comment]
maybe it’s worth adding it here

https://github.com/microsoft/vcpkg/blob/e1fc03c474a311ea658f5c2178c434c91776c9d9/scripts/cmake/vcpkg_find_acquire_program.cmake#L127

```
      set(POST_INSTALL_COMMAND ${CMAKE_COMMAND} -E remove python37._pth)
      find_program(PYTHON_EXECUTABLE NAMES python python3 python3.7 HINTS ${PATHS}  NO_DEFAULT_PATH)
        vcpkg_add_to_path(PREPEND ${PATHS})
      mark_as_advanced(PYTHON_EXECUTABLE)
```

and only specify this in the port files

`vcpkg_find_acquire_program(PYTHON3)`

**Comment by Neumann-A** [comment]
`vcpkg_find_acquire_program` should have options to append (APPEND_TO_PATH) or prepend (PREPEND_TO_PATH) to path automatically but that is a completely different issue

**Review by NancyLi1013 — APPROVED** [review]

**Comment by ras0219-msft** [comment]
LGTM. Thanks @LilyWangL, @NancyLi1013, and @Neumann-A!

**Pull request merged**